lib/elixir/lib/supervisor.ex

```diff
@@ -84,9 +90,177 @@ defmodule Supervisor do
   workers and supervisors as children, each of them with their specific
   configuration, shutdown values, and restart strategies.
 
-  The rest of this documentation will cover supervision strategies; also read
-  the documentation for the `Supervisor.Spec` module to learn about the
-  specification for workers and supervisors.
+  The rest of this document will cover how child processes are started,
+  how they can be specified, different supervision strategies and more.
+
+  ## The initialization process
+
+  In the previous section, we have started a supervisor with one child:
+
+      Supervisor.start_link([
+        {Stack, [:hello]}
+      ], strategy: :one_for_one)
+
+  The first argument given to `start_link` is a list of children.
+  In the example above, we have passed a tuple, where the child is
+  implemented by the `Stack` module and receives an initial argument
+  of `[:hello]` on `Stack.start_link/1`.
+
+  Generally speaking, starting the child process happens in three steps:
+
+    1. First the supervisor calls `Stack.child_spec([:hello])`. This
+       function must return a **child specification** which describes
+       how the `Stack` process is supervised. When you `use GenServer`,
+       a `child_spec/1` is automatically defined for you but we will see
+       when and how it can be configured. This function is called once
+       when the supervisor starts (or in case of hot code upgrades).
+
+    2. The **child specification** tells the supervisor which function
+       to invoke to start the child process. By default, it is the
+       `start_link/1` function, receiving the same argument and defined
+       in the same module as the `child_spec/1` function. This function
+       is called every time a new child process is necessary. For example,
+       when we crashed a `Stack` in the previous session,
+       `Stack.start_link([:hello])` was called once more to start a new stack.
+
+    3. Finally, `Stack.start_link/1` starts a new process that runs
+      `Stack.init/1`, which is responsible for setting a process that
+       will react to messages.
+
+  In summary, when the `Supervisor.start_link(children, opts)` is called,
+  it traverses the list of children and retrieves their `child_spec/1`.
+  Then each child specification contains how each child is started,
+  typically via the `start_link/1` function. The supervisor invokes the
+  `start_link/1` when it starts and whenever the child process crashes.
```

Comment:
`whenever the child process is restarted` rather than crashes - it might not be restarted on crash or might be restarted manually.